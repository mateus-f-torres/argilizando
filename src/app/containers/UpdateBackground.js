import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { displayBackground, lockBackground } from '../actions';
import PlayerBackground from '../components/PlayerBackground.jsx';

const mapStateToProps = (state) => {
  return {
    chosen: state.background.chosen
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getBack: (background) => {
      dispatch(displayBackground(background));
    },
    lockBack: (background) => {
      dispatch(lockBackground(background));
    }
  };
};

const UpdateBackground = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(PlayerBackground));

export default UpdateBackground;
