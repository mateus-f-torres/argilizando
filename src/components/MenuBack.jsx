//@flow
import * as React from 'react';
import {Link, Route, Switch} from 'react-router-dom';
import CardBack from './CardBack.jsx';
import BACKGROUNDS from '../data/backgrounds.js';
import type {Background} from '../types/props.js';

import 'Styles/MenuBack.scss';

type Props = {
  getBack: (string) => void,
  lockBack: ({}) => void,
  back: Background,
  match: {
    url: string,
  },
};

class MenuBack extends React.Component<Props> {
  showBack = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.getBack(e.currentTarget.name);
  }

  lockBack = () => {
    this.props.lockBack(this.props.back);
  }

  render() {
    return (
      <section id="BACK">
        <Switch>

          <Route exact path="/background" render={()=>
            <ul className="selection">
              <div id="title-box">
                <h2>Player Background</h2>
                <div className="horizontal-line no-bottom" />
              </div>
              {
                BACKGROUNDS.map((category, i) => {
                  let [type, options] = [...category];

                  return (
                    <div key={i}>
                      <h3>{type}</h3>
                      <div className="horizontal-line no-bottom" />
                      <ul>
                        {
                          options.map((back) => (
                            <li key={back.id}>
                              <Link to={this.props.match.url + '/' + back.id}>
                                <button name={back.id} onClick={this.showBack}
                                  id={
                                    back.id === 'polymorphed'
                                      ? 'long'
                                      : null
                                  }>
                                  {back.id}
                                </button>
                              </Link>
                            </li>
                          ))
                        }
                      </ul>
                    </div>
                  );
                })
              }
            </ul>
          }/>

          <Route path={this.props.match.url + '/:backgroundID'}
            render={()=>
              <CardBack lockBack={this.lockBack} {...this.props.back}/>
            }/>
        </Switch>
      </section>
    );
  }
}

export default MenuBack;
