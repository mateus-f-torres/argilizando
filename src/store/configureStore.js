import {createStore} from 'redux';
import rootReducer from '../reducers/index.js';

export default function configureStore() {
  return createStore(rootReducer);
}
