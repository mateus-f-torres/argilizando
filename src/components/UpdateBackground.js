//@flow
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayBackground} from 'Store/backgrounds/actions';
import {lockBackground} from 'Store/done/actions';
import MenuBack from 'Components/UpdateBackground/MenuBack.js';
import type {State} from 'Types/state.js';

const mapStateToProps = (state: State) => {
  return {
    back: state.background,
  };
};

const mapDispatchToProps = (dispatch: *) => {
  return {
    getBack: (background) => {
      dispatch(displayBackground(background));
    },
    lockBack: (background) => {
      dispatch(lockBackground(background));
    },
  };
};

const UpdateBackground = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuBack));

export default UpdateBackground;
