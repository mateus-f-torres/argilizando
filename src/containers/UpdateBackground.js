//@flow
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayBackground, lockBackground} from '../actions';
import MenuBack from '../components/MenuBack.jsx';
import type {State} from '../types/state.js';

const mapStateToProps = (state: State) => {
  return {
    back: state.background,
  };
};

const mapDispatchToProps = (dispatch: *) => {
  return {
    getBack: (background) => {
      dispatch(displayBackground(background));
    },
    lockBack: (background) => {
      dispatch(lockBackground(background));
    },
  };
};

const UpdateBackground = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuBack));

export default UpdateBackground;
