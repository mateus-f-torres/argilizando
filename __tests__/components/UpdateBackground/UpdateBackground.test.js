import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';

import configureStore from 'redux-mock-store';
import {mount} from 'enzyme';

import UpdateBackground from 'Components/UpdateBackground.js';
import MenuBack from 'Components/UpdateBackground/MenuBack.js';

const mockState = {
  background: {},
};

const mockStore = configureStore();
const store = mockStore(mockState);

describe('UpdateBackground', () => {
  beforeEach(() => {
    store.clearActions();
  });

  const container = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={
        [{pathname: '/background', key: 'testKey'}]}>
        <UpdateBackground />
      </MemoryRouter>
    </Provider>
  );

  // TODO check lockBackground dispatch

  describe('Unit Test', () => {
    test('render MenuBack component', () => {
      expect(container.containsMatchingElement(<MenuBack />)).toBe(true);
    });

    test('receive store.back', () => {
      const component = container.find('MenuBack');

      expect(component.prop('back')).toEqual(mockState.background);
    });

    test('dispatch getBack', () => {
      const component = container.find('MenuBack');

      const acolyteBtn =
        component.find('button').filter('[name="acolyte"]');

      const expectedActions = [
        {
          'type': 'DISPLAY_CHOSEN_BACKGROUND',
          'payload': 'acolyte',
        },
      ];

      acolyteBtn.simulate('click');
      expect(store.getActions()).toEqual(expectedActions);
    });

    test('dispatch lockClass', () => {});
  });

  describe('Snapshot Test', () => {
    expect(container).toMatchSnapshot();
  });
});
