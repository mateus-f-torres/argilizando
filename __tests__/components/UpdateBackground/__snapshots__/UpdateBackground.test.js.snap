// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/background",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/background",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/background",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(MenuBack))>
        <Route>
          <Connect(MenuBack)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/background",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/background",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/background",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <MenuBack
              back={Object {}}
              getBack={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/background",
                      "search": "",
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/background",
                    "search": "",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/background",
                  "search": "",
                }
              }
              lockBack={[Function]}
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                id="BACK"
              >
                <Switch>
                  <Route
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/background",
                        "url": "/background",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/background",
                        "search": "",
                      }
                    }
                    path="/background"
                    render={[Function]}
                  >
                    <ul
                      className="selection"
                    >
                      <div
                        id="title-box"
                      >
                        <h2>
                          Player Background
                        </h2>
                        <div
                          className="horizontal-line no-bottom"
                        />
                      </div>
                      <div
                        key="0"
                      >
                        <h3>
                          Literate
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="acolyte"
                          >
                            <Link
                              replace={false}
                              to="//acolyte"
                            >
                              <a
                                href="//acolyte"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="acolyte"
                                  onClick={[Function]}
                                >
                                  acolyte
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="heretic"
                          >
                            <Link
                              replace={false}
                              to="//heretic"
                            >
                              <a
                                href="//heretic"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="heretic"
                                  onClick={[Function]}
                                >
                                  heretic
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="noble"
                          >
                            <Link
                              replace={false}
                              to="//noble"
                            >
                              <a
                                href="//noble"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="noble"
                                  onClick={[Function]}
                                >
                                  noble
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="sage"
                          >
                            <Link
                              replace={false}
                              to="//sage"
                            >
                              <a
                                href="//sage"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="sage"
                                  onClick={[Function]}
                                >
                                  sage
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="1"
                      >
                        <h3>
                          Worker
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="charlatan"
                          >
                            <Link
                              replace={false}
                              to="//charlatan"
                            >
                              <a
                                href="//charlatan"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="charlatan"
                                  onClick={[Function]}
                                >
                                  charlatan
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="criminal"
                          >
                            <Link
                              replace={false}
                              to="//criminal"
                            >
                              <a
                                href="//criminal"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="criminal"
                                  onClick={[Function]}
                                >
                                  criminal
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="entertainer"
                          >
                            <Link
                              replace={false}
                              to="//entertainer"
                            >
                              <a
                                href="//entertainer"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="entertainer"
                                  onClick={[Function]}
                                >
                                  entertainer
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="guild"
                          >
                            <Link
                              replace={false}
                              to="//guild"
                            >
                              <a
                                href="//guild"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="guild"
                                  onClick={[Function]}
                                >
                                  guild
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="sailor"
                          >
                            <Link
                              replace={false}
                              to="//sailor"
                            >
                              <a
                                href="//sailor"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="sailor"
                                  onClick={[Function]}
                                >
                                  sailor
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="soldier"
                          >
                            <Link
                              replace={false}
                              to="//soldier"
                            >
                              <a
                                href="//soldier"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="soldier"
                                  onClick={[Function]}
                                >
                                  soldier
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="2"
                      >
                        <h3>
                          Stranger
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="dead"
                          >
                            <Link
                              replace={false}
                              to="//dead"
                            >
                              <a
                                href="//dead"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="dead"
                                  onClick={[Function]}
                                >
                                  dead
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="polymorphed"
                          >
                            <Link
                              replace={false}
                              to="//polymorphed"
                            >
                              <a
                                href="//polymorphed"
                                onClick={[Function]}
                              >
                                <button
                                  id="long"
                                  name="polymorphed"
                                  onClick={[Function]}
                                >
                                  polymorphed
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="hermit"
                          >
                            <Link
                              replace={false}
                              to="//hermit"
                            >
                              <a
                                href="//hermit"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="hermit"
                                  onClick={[Function]}
                                >
                                  hermit
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="outlander"
                          >
                            <Link
                              replace={false}
                              to="//outlander"
                            >
                              <a
                                href="//outlander"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="outlander"
                                  onClick={[Function]}
                                >
                                  outlander
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="3"
                      >
                        <h3>
                          Remarkable
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="folk"
                          >
                            <Link
                              replace={false}
                              to="//folk"
                            >
                              <a
                                href="//folk"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="folk"
                                  onClick={[Function]}
                                >
                                  folk
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="legendary"
                          >
                            <Link
                              replace={false}
                              to="//legendary"
                            >
                              <a
                                href="//legendary"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="legendary"
                                  onClick={[Function]}
                                >
                                  legendary
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="retired"
                          >
                            <Link
                              replace={false}
                              to="//retired"
                            >
                              <a
                                href="//retired"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="retired"
                                  onClick={[Function]}
                                >
                                  retired
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="urchin"
                          >
                            <Link
                              replace={false}
                              to="//urchin"
                            >
                              <a
                                href="//urchin"
                                onClick={[Function]}
                              >
                                <button
                                  id={null}
                                  name="urchin"
                                  onClick={[Function]}
                                >
                                  urchin
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                    </ul>
                  </Route>
                </Switch>
              </section>
            </MenuBack>
          </Connect(MenuBack)>
        </Route>
      </withRouter(Connect(MenuBack))>
    </Router>
  </MemoryRouter>
</Provider>
`;
